### PR TITLE
Partially fixed visibility for symbols (Ops, Nodes, Transformations, Matchers)

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Partially resolved issues with RTTI and AppleClang
Exports most of ngraph Ops which allows to use them from OpenCV, IE.
Note: it does not solve all the issues since we need to review NGraph API more carefully. I will submit more MR related to this topic after.

CC @ilyachur

### CODE REVIEW

File: src/ngraph/op/constant.hpp [hunk 1]
```diff
@@ -350,7 +349,6 @@ namespace ngraph
         class ScalarConstantLikeBase : public Constant
```
Comment (ilya-lavrenov) [inline src/ngraph/op/constant.hpp]: add `NGRAPH_API` here

File: src/ngraph/op/experimental/dyn_reshape.hpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ namespace ngraph
             /// "Converts" an input tensor into a new shape with the same number of elements.
             /// This op does not touch the actual data. If needed, use Transpose for that purpose.
             ///
-            class DynReshape : public Op
+            class NGRAPH_API DynReshape : public Op
             {
             public:
                 NGRAPH_API
```
Comment (ilya-lavrenov) [inline src/ngraph/op/experimental/dyn_reshape.hpp]: remove explicit `NGRAPH_API`

File: src/ngraph/op/passthrough.hpp [hunk 3]
```diff
@@ -31,11 +31,11 @@ namespace ngraph
...
-        class Passthrough;
+        class NGRAPH_API Passthrough;
     }
 }
 
-class ngraph::op::Passthrough final : public Op
+class NGRAPH_API ngraph::op::Passthrough final : public Op
 {
 public:
     NGRAPH_API
```
Comment (ilya-lavrenov) [inline src/ngraph/op/passthrough.hpp]: remove explicit `NGRAPH_API`

File: src/ngraph/op/passthrough.hpp [hunk 4]
```diff
@@ -31,11 +31,11 @@ namespace ngraph
         /// N.B. Not all backends support all operation languages; a
         /// given backend might only support a given passthrough
         /// operation language in certain modes.
-        class Passthrough;
+        class NGRAPH_API Passthrough;
```
Comment (ilya-lavrenov) [inline src/ngraph/op/passthrough.hpp]: remove

File: src/ngraph/frontend/onnx_import/core/null_node.hpp [hunk 5]
```diff
@@ -33,7 +33,7 @@ namespace ngraph
         ///
         /// More:
         /// https://github.com/onnx/onnx/blob/master/docs/IR.md#optional-inputs-and-outputs
-        class NullNode : public ngraph::Node
+        class NGRAPH_API NullNode : public ngraph::Node
```
Comment (tomdol) [inline src/ngraph/frontend/onnx_import/core/null_node.hpp]: This change causes linker errors in ONNXRuntime. I've sent more details over email but I think this class can't use the `NGRAPH_API` decorator.

Comment (ilya-lavrenov) [reply src/ngraph/frontend/onnx_import/core/null_node.hpp]: Reverted the change.

File: src/ngraph/frontend/onnx_import/core/null_node.hpp [hunk 6]
```diff
@@ -36,6 +36,7 @@ namespace ngraph
         class NullNode : public ngraph::Node
         {
         public:
+            NGRAPH_API
```
Comment (tomdol) [inline src/ngraph/frontend/onnx_import/core/null_node.hpp]: This occurence of the `NGRAPH_API` decorator causes another linker error.

**Comment by ilya-lavrenov** [comment]
It is hard to split into smaller pieces since changing one part leads to changing another one. originally, I wanted to create simple to review PR, but it become bigger :)

**Pull request merged**